WordPress/Classes/ViewRelated/System/Notices/NoticePresenter.swift

```diff
@@ -5,16 +5,31 @@ import WordPressFlux
 
 class NoticePresenter: NSObject {
     private let store: NoticeStore
-    private let presentingViewController: UIViewController
+    private let window: UntouchableWindow
+    private var view: UIView {
+        guard let view = window.rootViewController?.view else {
+            fatalError("Root view controller shouldn't be nil")
+        }
+        return view
+    }
     private var currentContainer: NoticeContainerView?
 
     let generator = UINotificationFeedbackGenerator()
 
     private var storeReceipt: Receipt?
 
-    private init(store: NoticeStore, presentingViewController: UIViewController) {
+    private init(store: NoticeStore) {
         self.store = store
-        self.presentingViewController = presentingViewController
+
+        let windowFrame: CGRect
+        if let mainWindow = UIApplication.shared.keyWindow {
+            windowFrame = mainWindow.offsetToAvoidStatusBar()
+        } else {
+            windowFrame = .zero
+        }
+        window = UntouchableWindow(frame: windowFrame)
+        window.windowLevel = .alert
+        window.makeKeyAndVisible()
```

Comment:
Why does this window need to be the key window?

Everything worked fine in my testing but I have come across issues before where somethings didn't work correctly if the main window wasn't key (I think there was some keyboard/first responder issues).

Reply:
I'm also curious why you decided to have the window always be present as opposed to showing/hiding it along with the alerts.

Reply:
I had no idea about using `window.isHidden` instead of `makeKeyAndVisible()`. Awesome.

I had tried to create/destroy the window as needed and it got awkward. TIL that I could hide it instead—that sounds great. Thanks @jtreanor! 👍